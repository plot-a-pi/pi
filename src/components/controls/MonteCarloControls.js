import React from 'react';
import PropTypes from 'prop-types';

const MonteCarloControls = ({ actions }) => (
  <section>
    {actions.map(({ name, text, actionCreator }) => (
      <button key={name} onClick={actionCreator}>
        {text || name}
      </button>
    ))}
  </section>
);

MonteCarloControls.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    text: PropTypes.string
  })).isRequired
};

export default MonteCarloControls;
