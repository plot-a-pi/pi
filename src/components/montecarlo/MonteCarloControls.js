import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../containers/MonteCarlo.css';

const MonteCarloControls = ({ actions }) => (
  <section className={styles.controls}>
    <h3>Add Darts</h3>
    {actions.map(({ name, text, actionCreator }) => (
      <button key={name} onClick={actionCreator}>
        {text || name}
      </button>
    ))}
  </section>
);

MonteCarloControls.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    text: PropTypes.string
  })).isRequired
};

export default MonteCarloControls;
