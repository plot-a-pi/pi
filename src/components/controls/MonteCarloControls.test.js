import React from 'react';
import { shallow } from 'enzyme';
import MonteCarloControls from './MonteCarloControls';

describe('MonteCarloControls component', () => {
  it('renders the MonteCarloControls', () => {
    const wrapper = shallow(<MonteCarloControls actions={[{ name: 'test-name', text: 'test-text', actionCreator: () => {} }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
