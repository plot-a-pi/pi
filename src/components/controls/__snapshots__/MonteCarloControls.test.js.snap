// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonteCarloControls component renders the MonteCarloControls 1`] = `
<section>
  <button
    key="test-name"
    onClick={[Function]}
  >
    test-text
  </button>
</section>
`;
